onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -650,6 +725,38 @@ void InitializeSession(InferenceSession* sess, const std::vector<std::string>& p
   OrtPybindThrowIfError(sess->Initialize());
 }
 
+bool CheckIfTensor(const InputDefList& def_list,
+                   const std::string& name,
+                   /*out*/ onnx::TypeProto& type_proto) {
+  auto ret_it = std::find_if(std::begin(def_list), std::end(def_list),
+                             [&name](const NodeArg* node_arg) { return name == node_arg->Name(); });
+  if (ret_it == std::end(def_list)) {
+    throw std::runtime_error("Failed to find NodeArg with name: " + name + " in the def list");
+  }
+
+  const auto* temp = (*ret_it)->TypeAsProto();
+  if (!temp) {
+    throw std::runtime_error("Corresponding type_proto is null");
+  } else {
+    type_proto = *temp;
+  }
+
+  return type_proto.has_tensor_type();
+}
+
+bool IsNumericNumpyType(int npy_type) {
+  return npy_type != NPY_STRING && npy_type != NPY_UNICODE && npy_type != NPY_VOID && npy_type != NPY_OBJECT;
+}
```

Comment:
Not sure if something limits the possibilities here. Would something like (npy_type < NPY_OBJECT ||  npy_type == NPY_HALF) be safer?

Reply:
yeah changed it to this - it makes sense